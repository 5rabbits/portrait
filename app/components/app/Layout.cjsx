React = require('react')

@Layout = React.createClass
  render: ->
    return (
      <section className="layout-wrapper" id="block-app">
        <header className="layout-header" id="block-header">
          { @props.header }
        </header>
        <nav className="layout-navigation" id="block-navigation">
          { @props.navigation }
        </nav>
        <main className="layout-main" id="block-main">
          { @props.main }
        </main>
      </section>
    )

module.exports = @Layout
