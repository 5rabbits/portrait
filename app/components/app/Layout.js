import React, { PureComponent } from 'react'

export default class Layout extends PureComponent {
  render() {
    return (
      <section className="layout-wrapper" id="block-app">
        <header className="layout-header" id="block-header">
          { this.props.header }
        </header>
        <nav className="layout-navigation" id="block-navigation">
          { this.props.navigation }
        </nav>
        <main className="layout-main" id="block-main">
          { this.props.main }
        </main>
      </section>
    );
  }
}
