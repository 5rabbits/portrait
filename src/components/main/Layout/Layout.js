import React from 'react'
import childrenPropType from 'propTypes/children'

const Layout = props =>
  <section className="layout-wrapper" id="block-app">
    <header className="layout-header" id="block-header">
      {props.header}
    </header>
    <nav className="layout-navigation" id="block-navigation">
      {props.navigation}
    </nav>
    <main className="layout-main" id="block-main">
      {props.main}
    </main>
  </section>

Layout.propTypes = {
  header: childrenPropType,
  main: childrenPropType,
  navigation: childrenPropType,
}

Layout.defaultProps = {
  header: null,
  main: null,
  navigation: null,
}

export default Layout
