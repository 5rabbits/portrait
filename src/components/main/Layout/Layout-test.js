import { React, toJSON, mount } from 'test/helper'
import Layout from 'main/Layout'

describe('Layout', () => {
  it('should render with minimum props', () => {
    const component = mount(<Layout />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
