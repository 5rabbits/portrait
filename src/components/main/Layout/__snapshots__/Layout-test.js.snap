// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render with minimum props 1`] = `
<Layout>
  <section
    className="layout-wrapper"
    id="block-app"
  >
    <header
      className="layout-header"
      id="block-header"
    />
    <nav
      className="layout-navigation"
      id="block-navigation"
    />
    <main
      className="layout-main"
      id="block-main"
    />
  </section>
</Layout>
`;
